crates/biome_js_analyze/src/lint/correctness/no_undeclared_dependencies.rs

```diff
@@ -34,32 +38,199 @@ declare_lint_rule! {
+                            description: Some("This type of dependency will be always available or unavailable.".to_owned()),
+                            ..Default::default()
+                        })),
+                        ..Default::default()
+                    }),
+                    Schema::Object(SchemaObject {
+                        instance_type: Some(InstanceType::Array.into()),
+                        array: Some(Box::new(ArrayValidation {
+                            items: Some(SingleOrVec::Single(Box::new(Schema::Object(SchemaObject {
+                                instance_type: Some(InstanceType::String.into()),
+                                ..Default::default()
+                            })))),
+                            min_items: Some(1),
+                            ..Default::default()
+                        })),
+                        metadata: Some(Box::new(Metadata {
+                            description: Some("This type of dependency will be available only if the linted file matches any of the globs.".to_owned()),
+                            ..Default::default()
+                        })),
+                        ..Default::default()
+                    })
+                ]),
+                ..Default::default()
+            })),
+            ..Default::default()
+        })
+    }
+}
+
+impl DependencyAvailability {
+    fn is_available(&self, path: &str) -> bool {
+        match self {
+            Self::Available => true,
+            Self::Unavailable => false,
+            Self::FilesGlob(globs) => globs.iter().any(|glob| glob.is_match(path)),
+        }
+    }
+}
+
+/// Rule's options
+#[derive(
+    Clone, Debug, Default, Deserializable, Eq, PartialEq, serde::Deserialize, serde::Serialize,
+)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct NoUndeclaredDependenciesOptions {
+    /// If set to `false`, then the rule will show an error when `devDependencies` are imported. Defaults to `true`.
+    #[serde(default)]
+    dev_dependencies: DependencyAvailability,
+
+    /// If set to `false`, then the rule will show an error when `peerDependencies` are imported. Defaults to `true`.
+    #[serde(default)]
+    peer_dependencies: DependencyAvailability,
+
+    /// If set to `false`, then the rule will show an error when `optionalDependencies` are imported. Defaults to `true`.
+    #[serde(default)]
+    optional_dependencies: DependencyAvailability,
+}
+
 impl Rule for NoUndeclaredDependencies {
     type Query = Manifest<AnyJsImportLike>;
     type State = ();
     type Signals = Option<Self::State>;
-    type Options = ();
+    type Options = NoUndeclaredDependenciesOptions;
 
     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
         let node = ctx.query();
         if node.is_in_ts_module_declaration() {
             return None;
         }
 
+        let path = ctx.file_path().to_str()?;
```

Comment:
If we use `Path` as suggested in a previous comment, we can remove this validation:
```suggestion
        let path = ctx.file_path();
```